orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/tasks/providers/aws/cloudformation/WaitForCloudFormationCompletionTask.java

```diff
@@ -72,8 +72,15 @@ public TaskResult execute(@Nonnull Stage stage) {
           isChangeSet
               ? getChangeSetInfo(stack, stage.getContext(), "status")
               : getStackInfo(stack, "stackStatus");
-      if (isComplete(status) || isEmptyChangeSet(stage, stack)) {
+      if (isComplete(status)) {
         return TaskResult.builder(ExecutionStatus.SUCCEEDED).outputs(stack).build();
+      } else if (isEmptyChangeSet(stage, stack)) {
```

Comment:
Again, apologies - I don't know much about CF... I assume, but want to confirm: Will `isComplete` return false in case of an empty changeset? Or should the `if isEmptyChangeSet...` be before `if isCompelte`?

Reply:
yes it will return false as an empty changeset shows as `FAILED` in cloud formation terms. It definitely needs to go before the `isFailed` though. Maybe this part is calling for a refactor in a later PR.